### PR TITLE
Feature/consortium test

### PR DESCRIPTION
**Pull request by jaybutera**
- Bypass singleton pattern, CommitteeSet generates new ConsortiumSet or gets deployed contract from chainspec based on conditional setting directly. Any sub-contracts needing a reference to the set gets it from the parent CommitteeSet contract. This allows parameters to be passed into contract during testing phase and avoids choosing between singleton or factory by decoupling the process.

- Conditional parameter to removeSupport specifies whether or not the method will have the "removeValidator" side effect. This was causing recursive bugs. Since the method is private, the extra parameter comes at no cost of usability to the user. Could also be seen as a curried function to pattern match two functions. Makes me feel better with that perspective.

- Could only add one validator to pending list per finalize. Fixed by removing whenFinalized modifier. Thoughts?

- Modifiers use requires to force revert on failure that may otherwise go unchecked

- newStatus returns ValidatorStatus storage pointer. Should make the process unambiguous as to what happened during the call.

- Few minor bug fixes that don't change intended functionality

### CODE REVIEW

**Comment by davesag** [comment]
Looks like you need to fix the linting issues

File: contracts/BurnableStakeBank.sol [hunk 1]
```diff
@@ -37,6 +37,7 @@ contract BurnableStakeBank is Lockable {
     Checkpoint[] public stakeHistory;
     Checkpoint[] public burnHistory;
     uint256 minimumStake;
+    address internal constant SYSTEM_ADDRESS = 0x00fffffffffffffffffffffffffffffffffffffffe;
```
Comment (tavakyan) [inline contracts/BurnableStakeBank.sol]: Should we just delete this line? I don't think its used in in this contract or in the inheritance hierarchy.

File: contracts/CommitteeSet.sol [hunk 2]
```diff
@@ -21,19 +21,31 @@ pragma solidity ^0.4.24;
...
 import './ConsortiumSetSingleton.sol';
 import './PublicSetSingleton.sol';
+import './ChainSpec.sol';
 
 
 // @title Contract to create committee from consortium and public validators
 // @notice Committees change every dynasty
 contract CommitteeSet is SystemValidatorSet {
-    SystemValidatorSet private consortiumSet = ConsortiumSetSingleton.instance();
+    SystemValidatorSet private consortiumSet;// = ConsortiumSetSingleton.instance();
```
Comment (tavakyan) [inline contracts/CommitteeSet.sol]: This is fine, you can probably just remove comment

File: contracts/BurnableStakeBank.sol (hunk 1 above)
Comment (jaybutera) [reply contracts/BurnableStakeBank.sol]: Huh thought I just took this out.. Just committed these two changes.

File: contracts/ConsortiumSet.sol [hunk 3]
```diff
@@ -55,6 +55,7 @@ contract ConsortiumSet is SystemValidatorSet, InitialConsortiumSet {
     uint internal constant maxValidators = 60;
     address[] private validatorsList;
     mapping(address => ValidatorStatus) private validatorsStatus;
+    address[] internal pendingList;
```
Comment (tavakyan) [inline contracts/ConsortiumSet.sol]: I think we still gotta figure out how initialize the pending list, but perhaps thats just the next step.  Originally I was inheriting it because I wanted to template it out. If we pass it in from constructor is it going to be hard coded in committee set when we init? Hoping there is a simple way to do this for the chainspec.

File: contracts/ConsortiumSet.sol [hunk 4]
```diff
@@ -239,38 +269,49 @@ contract ConsortiumSet is SystemValidatorSet, InitialConsortiumSet {
...
+            validatorsStatus[validator] = ValidatorStatus({
+                isValidator: false,
+                index: pendingList.length,
+                support: AddressVotes.Data({ count: 0 }),
+                supported: new address[](0),
+                benignMisbehaviour: AddressVotes.Data({ count: 0 })
+            });
+        }
+
+        return validatorsStatus[validator];
```
Comment (tavakyan) [inline contracts/ConsortiumSet.sol]: You ended up getting this working?

File: contracts/ConsortiumSet.sol (hunk 3 above)
Comment (jaybutera) [reply contracts/ConsortiumSet.sol]: Ya so for testing, the committee set takes a list of initial consortium members. I think that makes sense as a design in plain English. When reading from the chainspec, that input parameter is ignored (can be empty) and instead pulls the deployed contract which will have the pendingList passed into the contract from the chainspec bytecode.

File: contracts/ConsortiumSet.sol (hunk 4 above)
Comment (jaybutera) [reply contracts/ConsortiumSet.sol]: ya the problem was the modifier was using an if-conditional execution instead of require. Wasn't catching the error

**Comment by davesag** [comment]
@jaybutera I think changes to linting rules ought to be discussed before taking unilateral action as those changes need to be applied to all reveal projects.  While I don't disagree with your change, it's a general principle of team-based development that all such changes need to be agreed to by all developers before being enacted.

Also why was this commit approved and merged, given the CI is still reporting failures?
